### ISSUE TITLE
[QA] Regression Testing Cycle - V0.4.0

### ISSUE THREAD

**Issue by kjnanda** [GoogleChromeLabs/ps-analysis-tool#339]
# QA -> develop branch

-------

-  [X] **1. There seems to be a problem with the toggle column feature. When attempting to toggle the column, errors are encountered on the chrome://extension page. Additionally, when using the 'Toggle All' function to revert the changes, the columns do not render properly**.

> [!Important]
This issue is present on `feat/column-reordering` branch. Since it was checked out from dev, I checked on dev branch as well. The issue presents there as well.

Steps to replicate the issue -> https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/902c9c26-e72a-435e-b71e-2029ccde5478

|  | Screenshot|
|--- |--- |
| chrome://extensions | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/a017848d-45bb-464c-b004-2e882b17c237) |
| UI glitch on PSAT  | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/82fccb28-a491-49f0-9b08-1852e6fb1fca) |

-------

- [x] **2. The cookie icon show with white color when clicked else where**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/55ce851e-a316-4a30-8b6a-a214b56f36d7)</kbd>

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/4797181c-f59d-4cf8-bc96-312da6ec5767

----

- [x] **3. I need to click the 'Analyze this tab' button multiple times to retrieve the cookies, and occasionally, it requires more than two clicks. Additionally, there is a noticeable flickering effect that disrupts the basic functionality of the PSAT.**

| With older codebase | With latest code pull from dev branch (commit - 702e0699b2f37dae4f9b783c05cbb606541083d5) |
|---- |------ |
| https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/831f9db3-945f-44d8-aea2-65f84a038999 | https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e46b2694-a42c-42bb-9853-06880193108b |

Errors on chrome://extensions page - 
<kbd><img width="692" alt="Screenshot 2023-12-20 at 12 12 03" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e341e746-6778-4949-ae6e-b0a5e6a8cf27"></kbd>

-----

-  ~**4. Delete single cookie icon - it should only enabled after any row is selected**~ Not in a scope for current release

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/4b072411-a28a-4f53-8131-16a456cafd45

---

-  **5. UI glitch on Toggle column** [Ref:comment ](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/339#issuecomment-1863886942)

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/05c1deeb-2859-4464-a1d3-7aa4a8375b24)</kbd>

---

- [x] **6. Warnings are not being shown for the cookies that are going to be affected**

| PSAT | Application Tab with console output showing Affected Cookies |
|---- |---- |
| <kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ce1a3627-083d-45c3-a267-1db823b23067)</kbd> | <kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e18b54b7-fa50-4fa3-87d2-5c0ae45b57df)</kbd>

----

- [x] **7. Sidebar : Frame Selection Issue**

Tab 1: The user navigates to any frame to view the table data.
Tab 2 : It will show `Analyse this tab` button. Clicking on this button it will load the cookies.
Tab 1: Return to Tab 1, where you will find the correct display of 'Analyse this tab.' Clicking on it successfully loads the cookie for the website. However, it is important to note that there is an issue in the sidebar – the 'Cookies Landing' frame should be selected, but currently, it is not reflecting this behavior.

| Expected behaviour | Actual behaviour |
|----- |------ |
| ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/108fcf3f-22db-4cda-b500-821f5359077b) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/72b4794b-14a4-4bdd-b564-87c71139e62e) |

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/e39e74ff-bc97-4b86-88f9-18b88c13853d

------

- **8. As per the latest implementation the cookie jar should show `0` count in pop up** - Invalid issue

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/34089d63-7e0b-4618-a9cb-42f1248607c3)</kbd>

----

- [x] **9. The filter's basic functionality is not working as expected**

- filter disappears once you click on the filter option for which the cookie is not present
- filter icon gets disabled 
- clearing the filter option from top bar, retains the filter but the selected options does nor come up

Please refer to the screencast below.

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c5a93043-c7dc-42f0-a0c9-f6ea02212f86

---

- [x] **10. Filters: Missing padding from top and bottom when the single filter is applied.**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/a3e8f838-5ea8-4768-8f4c-0318d3736154)</kbd>

----

- [x] **11. `Frame overlay icon` is not visible.** 

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/770cfb95-a8f3-4ad5-b691-717d0ea8aa4a)</kbd>

> [!Note]
The added fix is partially working. It does not show the frame overlay icon for the very first site. or until the user explicitly clicks on `Analyse this tab` button. 

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2ade8e15-9aec-4ed5-a3c6-d8d4fef4b202

----

-  ~**12. Clicking `X` (single cookie delete icon) without selecting any row throws the error on chrome://extension page**~ Not in scope of current release now. 

<kbd><img width="1465" alt="Screenshot 2023-12-18 at 18 43 56" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/7c180c1a-3bb3-4976-8df9-89abacf36b1d"></kbd>

-----

## Testing with latest commit - 702e0699b2f37dae4f9b783c05cbb606541083d5 from dev branch

- [x] **13. URL is not working as expected. rewriting the URL does not load the correct set of cookies within some expected set of time and takes longer than expected.**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/60979c31-552c-4406-9a4e-6fe2b59f49d4

----

- ~**14. There seems to have an issue with `Blocked Cookie`. Either it shows very less cookie or zero cookie while console tabs there are lot more cookies to be blocked**~ Being tracked separately via https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/354

Test Site - https://www.aljazeera.com/

| Attempt 1 | Attempt 2 |
|--- |---- |
| ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/5bfeaf9b-d414-47d1-bbcd-71d4200b7d7d) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/bcc486ad-f06a-452b-b38d-a5420f6b8c5b) |

Test Site - https://www.livemint.com/

<kbd><img width="1014" alt="Screenshot 2023-12-19 at 18 54 49" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c49b9457-18de-47b0-9d66-537b1ea33572"></kbd>

With latest code pull from dev branch 

| Tab | Screenshot |
|--- |--- |
| Network Tab | <img width="1467" alt="Screenshot 2023-12-20 at 14 24 12" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/98bb0092-ac93-47cc-a662-43d70d05b5bd"> |
|PSAT Table | <img width="1448" alt="Screenshot 2023-12-20 at 14 24 02" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c1f241f3-cdd3-48ec-91bb-4a0b65041898"> |
| Cookies Landing PAge | <img width="1450" alt="Screenshot 2023-12-20 at 14 22 47" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/ecd6f2b3-7c2f-4ebc-8092-fbada1522277"> |

-----

- [x] **15. Size and priority are not visible under filter**

<kbd><img width="1470" alt="Screenshot 2023-12-19 at 13 26 53" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/2de44e55-ec82-4266-9fa1-e49f9ca727c0"></kbd>

----

- [x] **16. There seems to have some jerk/flickering effect while opening any landing page or sub menus of it.**

| Visual | Link| 
|-- |-- |
|Screencast 1 | https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/9233d40a-0f45-471a-9933-1947ff578db1 |
| Screencast 2 | https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/a5a1c171-7e33-49a6-abd7-3824a118ba3c |
| While switching between the tabs it also flickers | https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/d63a4c94-7ee5-4f21-94a8-e35817eeca44 |
| Error log | <img width="659" alt="Screenshot 2023-12-19 at 16 57 59" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/7f7a045e-e84d-49c4-9057-968e7ea0d3c1"> |

----

- [x] **17. Please make the colors uniform for the links on landing pages**

On both the themes, there is a difference in colors of link but for light mode it is not much noticeable.

> [!Important]
Not a regression bug

| Dark Mode | Light Mode | 
|--- |--- |
| ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c643db02-1d9e-4aee-acc0-8f4a0561ce09) | ![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/dc80ee79-6ff4-4097-ae7f-7a66b684eec1) |

----

-  ~**18. Frame overlay: Selecting the filters from the dedicated sidebar does not reflect's in tooltip. If I select any filter for for the dedicated frame, then it should reflect the same in tooltip as well.**~ Invalid [Ref : comment ](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/339#issuecomment-1862737899)

Please check the below visual where I have tried to showcase
- How the current build is behaving and not changing the values
-  Also, showcased the same using  stable version how it should reflect 

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/027f9cad-72dd-4eec-b94c-79269ebe460f

**Comment by pavanpatil1** [comment]
- [ ] CLI: In the CSV and sitemap, Cookies count are not visible in the report. 

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/44057535/c461035f-ca69-4840-90fc-a52df96e48f7)

**Comment by amovar18** [comment]
> Warnings are not being shown for the cookies those going to be affected

Not able to reproduce this. It works fine on my system.

>  As per the latest implementation the cookie jar should show 0 count in pop up

Only for those tabs which are being analysed. If a tab is not being analysed then the count wont show.

**Comment by gagan0123** [comment]
**Describe the bug**
The cookies landing page that displays the number of blocked cookies randomly gets stuck at 0, despite the confirmation of blocked cookies in the Network Tab of Chrome's developer tools.

**To Reproduce**
Steps to reproduce the behavior:
1. Load the Privacy Sandbox Analysis Tool extension in Chrome.
1. Open a new instance and navigate to a website (e.g., example.com).
1. Close (if already open) and reopen the developer tools to ensure the Privacy Sandbox tab is present.
1. Observe the website, which likely involves third-party cookies, eg, https://edition.cnn.com .
1. Check the cookies landing page of the Privacy Sandbox Analysis Tool for the count of blocked cookies.
1. Navigate to the Network Tab and filter the results by blocked response cookies.
1. Notice the discrepancy between the actual blocked cookies in the Network Tab and the zero count displayed on the cookie's landing page.

**Expected behavior**
The expected behavior is that the cookies landing page counter of the Privacy Sandbox Analysis Tool accurately reflects the number of third-party cookies blocked by the browser, in accordance with the data shown in the Network Tab.

**Screenshots**

![Screenshot from 2023-12-19 04-42-08](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/1535505/757bb96d-f4f9-482c-a3c3-c800f4938ceb)

Screencast: 
https://www.youtube.com/watch?v=cCHg9rT5uyQ

**Desktop (please complete the following information):**
 - OS: Pop!_OS 22.04 LTS x86_64
 - Chrome Version: Version 120.0.6099.109 (Official Build) (64-bit)

**Additional context**
This issue seems to occur randomly and poses a significant problem in accurately debugging and analyzing third-party cookie behavior in Chrome.

**Comment by ayushnirwal** [comment]
Item 8 in the first comment is an invalid issue, since it is expected that pop-up badge should not appear when a website is not analyzed.

**Comment by ayushnirwal** [comment]
> 12. Clicking X (single cookie delete icon) without selecting any row throws the error on chrome://extension page.

This will not be addressed since we are not releasing cookie modification and deletion functionalities in this release

**Comment by amovar18** [comment]
Item 13 is also expected due to the usage of PROMISE QUEUE

**Comment by amovar18** [comment]
Frame overlay icon will only be visible once user clicks analyse this tab button if tab is not being analysed it wont be visible.

**Comment by ayushnirwal** [comment]
> Describe the bug
The cookies landing page that displays the number of blocked cookies randomly gets stuck at 0, despite the confirmation of blocked cookies in the Network Tab of Chrome's developer tools.

@gagan0123 Would you mind adding the information about any flags you have enabled ( third-party phaseout etc ) or you have blocked 3p cookies from the Chrome settings?

Also if this happens again please follow the following steps
- Right click on the devtools panel and select inspect frame

**Comment by ayushnirwal** [comment]
> After passing the sitemap file in the CLI. It is showing 0 URLs in the file even though there are URL's in the sitemap.xml file.

You are using the wrong argument its 
`npm run cli -- -p <path to xml>`

**Comment by fellyph** [comment]
A minor issue at the cookie table: when we right-click a cookie, the menu has `.` at the end of the `Copy network filter string` option:
![dot at the menu](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/330792/648c748c-8e01-40e6-89a0-ad9736fcd050)

**Comment by kjnanda** [comment]
@ayushnirwal Agree with @fellyph's suggestion. I have added the related issue [here](https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/320#issuecomment-1855302678) also under the subtask 3

**Comment by mayan-000** [comment]
> A minor issue at the cookie table: when we right-click a cookie, the menu has . at the end of the Copy network filter string option

Removing the period

@kjnanda @fellyph

**Comment by mayan-000** [comment]
> 5: UI glitch on Toggle column

Will be updating the logic in a later release, not planning to push any changes for now. 

cc @kjnanda

**Comment by kjnanda** [comment]
@Mayank Rana Size is visible in filter but can we please show it in ASC sorting order?

**Comment by kjnanda** [comment]
@ayushnirwal With the latest code pull from dev branch - Commit - 1d18c6d2035cec36ce690f63d401fa0efb70bf5e

The following issues are noticed:

- [ ] Only Top frame shows the cookies. And rest frames are blank

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/07cadda7-cd47-4d73-a5b3-855e242ed600

**Comment by kjnanda** [comment]
@ayushnirwal Below task needs attention from you. 

Subtask 3, 9 (partially fixed - https://github.com/GoogleChromeLabs/ps-analysis-tool/issues/339#issuecomment-1863939211 ), 13

**Comment by kjnanda** [comment]
## QA - branch `fix/blocked-cookies-2`

----

- [ ] **`Invalid Domain` option is missing under filters**

---

- [ ] **Popup -> Clicking on "analyse this tab" button is not functioning as expected and throws error on chrome://extensions page** 

<kbd>
<img width="685" alt="Screenshot 2023-12-22 at 12 11 05" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/61656321-ed8d-492f-8f2d-a86c478ff5ce">
</kbd>

**Comment by kjnanda** [comment]
## QA - `fix/blocked-cookie-count` 

- [ ] **1. Blocked cookie count is not visible for 2nd tab**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/58309494-5be6-458d-854d-0b50d8e2e9f3)</kbd>

**Comment by kjnanda** [comment]
## Testing against `fix/blocked-cookie-count` branch

-------

- [ ] **Cookie Blocked Reason : I see `NotOnPath` has 1 cookie as per cookies landing page. However, Could not see any cookies that falls under this category.**

Test Site : https://edition.cnn.com/

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c617c598-1dd7-44f8-9a39-c35954af1694

**Comment by kjnanda** [comment]
## CLI Dash board 

---

@ayushnirwal Could you please look into this issue?

- [ ] **Horizontal matrix showing frames count is not visible on CLI**

<kbd>![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/f86a180b-849a-4531-866f-22e0b3c1cfd6)</kbd>

**Comment by kjnanda** [comment]
Closing this issue as it is addressed in V0.4.0 release
